corehq/apps/enterprise/resumable_iterator_wrapper.py

```diff
@@ -0,0 +1,48 @@
+from itertools import islice
+
+
+class ResumableIteratorWrapper:
+    def __init__(self, sequence_factory_fn, get_element_properties_fn=None, limit=None):
+        self.limit = limit
+
+        # if a limit exists, increase it by 1 to allow us to check whether additional items remain at the end
+        padded_limit = limit + 1 if limit else None
+        self.original_it = iter(sequence_factory_fn(padded_limit))
```

Comment:
Why pass `padded_limit` to `sequence_factory_fn`? Based on the test cases, `sequence_factory_fn` ignores the parameter.

Reply:
If you're asking why we need a limit, the issue is that Tastypie needs some way to communicate the 'limit' it receives with the underlying limit used by Elasticsearch. If we can't communicate this to elasticsearch, it leads to situations like the API asking for 100 records and Elasticsearch fetching 5000, leading to 4900 unused records. Or the reverse, where the API asks for 5000 records,and Elasticsearch fetches them 100 at a time, leading to 50 calls to Elasticsearch rather than 1.

Ideally, we'd just pass the limit all the way down, but we don't have control over how the Paginator is instantiated or how it is called -- we can tell Tastypie to use a certain Paginator class, but it controls the instatiation and the call. What we can control is the 'objects' object that gets passed to the paginator. That object can't know about the limit yet, because the paginator is responsible for setting that limit. So the object needs a way to receive the limit prior to retrieving the results.

The important part here is that the iterator needs to be able to receive a limit parameter, even if it does nothing with it. In the case of the tests, they ignore that limit, but if the underlying sequence was an elasticsearch query, it would need access to that limit